File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        if (k == 0 || nums.length == 1) {
            return false;
        }
        int left = 0;
        int curr = 1;
        while (left < nums.length - 1) {
            if (nums[left] == nums[curr] && Math.abs(left - curr) <= k) {
                return true;
            }
            if (nums[left] != nums[curr] && Math.abs(left - curr) <= k) {
                if (curr < nums.length - 1) {
                    curr++;
                } else {
                    left++;
                    curr = left + 1;
                }
            } else {
                left++;
                curr = left + 1;
            }
        }
        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int, int> hashMap;
        for (int i = 0; i < n; i++) {
            if (hashMap.find(nums[i]) == hashMap.end()) {
                hashMap[nums[i]] = i;
            } else {
                if (abs(hashMap[nums[i]] - i) <= k) {
                    return true;
                }
                hashMap[nums[i]] = i;
            }
        }
        return false;
    }
};